pandas/formats/format.py

```diff
@@ -2042,7 +2042,12 @@ def get_result_as_array(self):
         """
 
         if self.formatter is not None:
-            return np.array([self.formatter(x) for x in self.values])
+            # na_rep takes precedence over the specified formatter
+            if self.na_rep is not None:
+                return np.array([self.formatter(x) if notnull(x)
+                                 else self.na_rep for x in self.values])
```

Comment:
@jreback I just looked at the datetime formatters but I don't know what you mean. Could you be more explicit? Thanks.